src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NodeBase.java

```diff
@@ -0,0 +1,152 @@
+package edu.greenblitz.tobyDetermined.Nodesssss;
+
+import edu.wpi.first.math.util.Units;
+
+import java.util.LinkedList;
+
+import static edu.greenblitz.tobyDetermined.RobotMap.TelescopicArm.Elbow.STARTING_ANGLE_RELATIVE_TO_GROUND;
+
+public class NodeBase  {
+    private NodeArm REST_ABOVE_BELLY;
+    private NodeArm INTAKE_GRAB_CUBE_POSITION;
+    private NodeArm CONE_HIGH;
+    private NodeArm CONE_MID;
+    private NodeArm CUBE_HIGH;
+    private NodeArm CUBE_MID;
+    private NodeArm LOW;
+    private NodeArm ZIG_HAIL;
+    private NodeArm INTAKE_GRAB_CONE_POSITION;
+    private NodeArm PRE_CONE_DROP;
+    private NodeArm POST_CONE_DROP;
+    private NodeArm currentNode;
+    private int i = 1;
+    private final LinkedList<NodeArm> list = new LinkedList<>();
+    private static NodeBase instance;
+    private final double TOLERANCE_ANGLE = Units.degreesToRadians(3);
+
+    private final double TOLERANCE_LENGTH = 0.04;//In Meters
+
+    public static NodeBase getInstance() {
+        init();
+        return instance;
+    }
+
+    public static void init() {
+        if (instance == null) {
+            instance = new NodeBase();
+        }
+    }
+
+    public int addToList(NodeArm nodeArm, LinkedList<NodeArm> list){
```

Comment:
If I understand correctly, `i` is the number of nodes in the list.
If that really is the case, why does this function return the newly-set value of `i`?
I see no reason an `add` method should do this. If you need that number, use another method to return it.

Reply:
i start as 0 and it is the index of every node, because of that every time i use i for a index i need to change it, and because of that the i is changing in addToList